tests/all/volumes.t

```diff
@@ -111,3 +111,37 @@ nodes.each do |node|
   USE_NODE node
   puts TEST "cat /var/log/kadalu/mgr.log"
 end
+
+# Tests for volume re-use usecase
+def create_volume_and_get_id(cmd)
+    out = TEST cmd
+    last_line = out.strip.split("\n")[-1]
+    last_line.split(":")[-1].strip
+end
+
+nodes.each do |node|
+  USE_NODE node
+  TEST "systemctl stop kadalu-mgr"
+  TEST "rm -rf /var/lib/kadalu/info"
+  TEST "systemctl start kadalu-mgr"
+end
+
+# Case 1
+# Create vol6 & delete it. Reuse the same path with volume-id
+USE_NODE nodes[0]
+vol_id = create_volume_and_get_id "kadalu volume create DEV/vol6 server1:/exports/vol6/s1 server2:/exports/vol6/s2 server3:/exports/vol6/s3 --auto-create-pool --auto-add-nodes --no-start"
+TEST "kadalu volume delete DEV/vol6 --mode=script"
+new_vol_id = create_volume_and_get_id "kadalu volume create DEV/vol6 server1:/exports/vol6/s1 server2:/exports/vol6/s2 server3:/exports/vol6/s3 --auto-create-pool --auto-add-nodes --no-start --volume-id=#{vol_id}"
+EQUAL vol_id, new_vol_id, "Test description"
+
+nodes.each do |node|
```

Comment:
All lines below not required, if you move tests before cleanup as commented above.